src/makim/cli.py

```diff
@@ -334,8 +353,30 @@ def run_app() -> None:
     # Add dynamically created commands to Typer app
     for name, args in targets.items():
         create_dynamic_command(name, args)
-
-    app()
+    try:
```

Comment:
thanks for working on that ... this is a suggestion to keep it simpler:

```diff
diff --git a/src/makim/cli.py b/src/makim/cli.py
index 1efa3cf..08ceb8a 100644
--- a/src/makim/cli.py
+++ b/src/makim/cli.py
@@ -332,6 +332,36 @@ def extract_root_config() -> Dict[str, str | bool]:
         'verbose': verbose,
     }
 
+def _get_command_from_cli() -> str:
+    """
+    Get the group and target from CLI.
+
+    This function is based on `root_args_values_count`.
+    """
+    params = sys.argv[1:]
+    command = ""
+    root_args_values_count = {
+        '--dry-run': 0,
+        '--file': 1,
+        '--help': 0,  # not necessary to store this value
+        '--verbose': 0,
+        '--version': 0,  # not necessary to store this value
+    }
+
+    try:
+        idx = 0
+        while idx < len(params):
+            arg = params[idx]
+            if arg not in root_args_values_count:
+                command = f"flag `{arg}`" if arg.startswith("--") else arg
+                break
+
+            idx += 1 + root_args_values_count[arg]
+    except Exception:
+        pass
+
+    return command
+
 
 def run_app() -> None:
     """Run the typer app."""
@@ -356,26 +386,19 @@ def run_app() -> None:
     try:
         app()
     except SystemExit as e:
-        error_code = 2
-        target_index = 1
-        if e.code == error_code:
-            for flag in ['--file', '--verbose', '--dry-run']:
-                if flag in sys.argv:
-                    target_index = max(
-                        target_index,
-                        sys.argv.index(flag) + (2 if flag == '--file' else 1),
-                    )
-
-            command_used = (
-                sys.argv[target_index] if target_index < len(sys.argv) else ''
-            )
-            available_cmds = [cmd.name for cmd in app.registered_commands]
-            suggestion = suggest_command(command_used, available_cmds)
-            typer.secho(
-                f"Command {command_used} not found.\
- Did you mean '{suggestion}'?",
-                fg=typer.colors.RED,
-            )
+        # code 2 means command not found
+        if e.code != 2:
+            raise e
+
+        command_used = _get_command_from_cli()
+
+        available_cmds = [cmd.name for cmd in app.registered_commands]
+        suggestion = suggest_command(command_used, available_cmds)
+
+        typer.secho(
+            f"Command {command_used} not found. Did you mean '{suggestion}'?",
+            fg=typer.colors.RED,
+        )
         raise e
```

so in this code, I moved the business logic that get the command to its own function.
also I change the condition for code error, so I could move the rest of the code 1 indentation level behind.

by the way, take a look into this video to understand why I change the if condition and move the rest of the code 1 indentation behind:

https://www.youtube.com/watch?v=CFRhGnuXG-4